core/src/main/java/io/questdb/cutlass/http/processors/LineHttpProcessorState.java

```diff
@@ -283,21 +284,26 @@ private long getErrorLogLineHi(LineTcpParser parser) {
     private Status handleCommitError(Throwable ex) {
         errorId = ERROR_COUNT.incrementAndGet();
         errorLine = -1;
-        LOG.critical()
-                .$('[').$(fd).$("] could not commit [table=").$(parser.getMeasurementName())
-                .$(", errorId=").$(ERROR_ID).$('-').$(errorId)
-                .$(", ex=").$(ex.getMessage())
-                .I$();
 
+        final Status status;
+        final LogRecord errorRec;
         error.put("commit error for table: ").put(parser.getMeasurementName());
         if (ex instanceof CairoException) {
             CairoException exception = (CairoException) ex;
             error.put(", errno: ").put(exception.getErrno()).put(", error: ").put(exception.getFlyweightMessage());
-            return exception.isAuthorizationError() ? Status.SECURITY_ERROR : Status.INTERNAL_ERROR;
+            errorRec = exception.isCritical() ? LOG.critical() : LOG.error();
```

Comment:
there is a split brain decision possible here. I feel if status is INTERNAL ERROR then log must be critical. However, status and criticality decision is split between seemingly unrelated conditions.

better condition for both could be:

```

if (exception.isAuthorizationError() && !exception.isCritical()) {
  status = Status.SECURITY_ERROR;
  errorRec =  LOG.error()
} else {
// internal & critical
}
```

Reply:
That is how i had it originally, then changed it to this version.
I was torn between the two.
Will change it back then.

Reply:
done